src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,204 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * Represents the main logic of the code when the user enters an input.
+ */
+public class Parser {
+    private Scanner scan;
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+    private String arr[];
+    private String input = "";
+
+    /**
+     * A constructor to create a parser object.
+     *
+     * @param scan A scanner that takes in the input from the user.
+     * @param storage A storage that deals with loading and saving files.
+     * @param taskList A list that keeps track of all tasks.
+     * @param ui An object that perform tasks related to interactions with user.
+     */
+    Parser(Scanner scan, Storage storage, TaskList taskList, Ui ui) {
+        this.scan = scan;
+        this.storage = storage;
+        this.taskList = taskList;
+        this.ui = ui;
+    }
+
+    /**
+     * Checks if a string is an integer or not.
+     *
+     * @param s A string from the user input.
+     * @return True if it is an integer, false otherwise.
+     */
+    public static boolean isNumeric(String s) {
+        try {
+            Integer.parseInt(s);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Makes sense of the user input.
```

Comment:
I think this can be changed into something more active; ie: converts the user's inputs.